durable-storage/src/database.rs

```diff
@@ -0,0 +1,125 @@
+// SPDX-FileCopyrightText: 2025 Trilitech <contact@trili.tech>
+//
+// SPDX-License-Identifier: MIT
+
+use std::sync::Arc;
+
+use crate::merkle_layer::Key;
+use crate::merkle_layer::MerkleLayer;
+use crate::merkle_layer::MerkleLayerError;
+use crate::persistence_layer::PersistenceLayer;
+use crate::persistence_layer::PersistenceLayerError;
+
+#[derive(Clone)]
+#[expect(dead_code, reason = "Implemented in RV-827")]
+pub struct Database {
+    persistent: Arc<PersistenceLayer>,
+    merkle: Arc<MerkleLayer>,
```

Comment:
This should be a `MerkleWorker`. Otherwise, all operations would be synchronous.

```suggestion
    merkle: Arc<MerkleWorker>,
```